sbi/inference/posterior.py

```diff
@@ -565,6 +571,19 @@ def _ensure_single_x(x: Tensor) -> None:
                 """
             )
 
+    def _warn_if_posterior_was_focused_on_different_x(self, x: Tensor):
+        """Warn if user provides an x not equal to the x_o used during inference."""
+
+        if self._num_trained_rounds > 1 and (self._x_o_training_focused_on != x).any():
+            warn(
+                f"The posterior was trained over multiple rounds focused on a specific"
+                f" observation x_o={self._x_o_training_focused_on.tolist()}. The"
+                f" observation you provided x={x.tolist()} is not identical to x_o,"
+                f" which can lead to poor performance of the inference method."
+                f" Consider running inference with num_round==1, which allows to pass"
```

Comment:
`num_rounds` (plural and backticks) - backticks also in docstring, and in general when citing variables that exist in code.